teb_local_planner/test/homotopy_class_planner_test.cpp

```diff
@@ -36,14 +40,45 @@ class HomotopyClassPlannerTest : public teb_local_planner::HomotopyClassPlanner
     teb_local_planner::TebConfig cfg;
 };
 
-TEST(test, test) {
+class VisualizationSubscriber : public rclcpp::Node {
+    public:
+    VisualizationSubscriber() : rclcpp::Node("visualization_subscriber") {
+      teb_poses_sub_ = this->create_subscription<geometry_msgs::msg::PoseArray>("teb_poses", 1, std::bind(&VisualizationSubscriber::tebPosesCallback, this, std::placeholders::_1));
+    }
+
+    geometry_msgs::msg::PoseArray teb_poses;
+    bool teb_poses_fetched;
+    void tebPosesCallback(const geometry_msgs::msg::PoseArray::SharedPtr input_poses) {
+      if (input_poses) teb_poses = *input_poses;
```

Comment:
if (input_poses) does what exactly?

Reply:
checks if you received a null pointer but I guess we can trust ROS